### PR TITLE
fix(Scripts/Blackwing Lair): Supression traps should not be disarmed …

### LINKED ISSUE #11058: [BWL] Supression Room Traps should stay down once Broodlord Lashlayer is defeated
### Current Behaviour

Traps will keep coming up after Broodlord Lashlayer is defeated.

### Expected Blizzlike Behaviour

During fight traps should respawn with a random timer,
after fight all traps that got disarmed should stay down and any new traps disarmed should also stay down

### Source

_No response_

### Steps to reproduce the problem

.go c 84387
disarm traps kill dude watch respawn

### Extra Notes

_No response_

### AC rev. hash/commit

b01aa11f1803

### Operating system

Ubuntu

### Custom changes or Modules

mod-anticheat (master)  (conf disabled)
mod-autobalance (master)
mod-bg-item-reward (master)
mod-cfbg (master)
mod-cta-switch (master
mod-chromiexp (master)
mod-chromiecraft (master)
mod-desertion-warnings (master)
mod-duel-reset (master)
mod-eluna-lua-engine (master)
mod-ip-tracker (master)
mod-low-level-arena (master)
mod-multi-client-check (master)
mod-progression-system (master)
mod-pvp-titles (master)
mod-pvpstats-announcer (master)
mod-queue-list-cache (master) 
mod-rdf-expansion (master) 
mod-save-inventory (master)
mod-server-auto-shutdown (master)
mod-spellsystem-extended (master)
Acore_eventScripts (master)
Acore_SendAndBind (master)
Acore_TempAnnouncements (master)
Acore_ZoneDebuff (master) 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/106210503-bwl-supression-room-traps-should-stay-down-once-broodlord-lashlayer-is-defeated?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
…after Broodlord defeat.

Fixes #11058

## Issues Addressed:

- Closes #11058

## Tests Performed:

- Not tested.

## How to Test the Changes:

`.go c 84387`
disarm traps kill dude watch respawn

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Lynethris** [comment]
Not work. After that Broodlord die, all traps are disarm.

WoWScrnShot_031922_164133

**Comment by UltraNix** [comment]
@Lynethris Read again the ticket.

**Comment by Nefertumm** [comment]
Hmm, only those who were disarmed should be stay down, the other ones should stay up

**Comment by UltraNix** [comment]
@Lynethris I apologize.
Could you re-test it with the newest version?

**Comment by Lynethris** [comment]
Tested and work : 

- [X] Before the Broodlord defeat, all traps are re-armed after X time
- [X] When Broodlord die, all disarmed traps stay down
- [X] When Broodlord die, the new disarmed traps stay down

there is a situation where I don't know if we are good or not : 

- [ ] When server restart/crash, all trash are disarmed

@temperrr @Nefertumm an idea ?

**Comment by temperrr** [comment]
Last one is ATM the only way to deal with a crash so that's good

**Comment by UltraNix** [comment]
@Lynethris About the 4th problem - please, re-test it.

**Comment by Lynethris** [comment]
@UltraNix with the last commit, all traps are up, even disarmed traps before the server restart/crash.

What I wanted to say before is that I have no idea about the situation after a crash/server on blizzard server (and it's not mentioned on the issue). Actually, on master branch (and before the last commit), when server restart/crash, if Broodlord is alive, all traps are up. If he is dead, all traps are disarmed. It's maybe not a bug.

**Comment by UltraNix** [comment]
Okay. I revert that commit and leave as it was.

File: src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp [hunk 1]
```diff
@@ -75,6 +76,16 @@ class boss_broodlord : public CreatureScript
...
         }
 
+        void JustDied(Unit* /*killer*/) override
+        {
+            _JustDied();
+
+            std::list<GameObject*> _goList;
+            GetGameObjectListWithEntryInGrid(_goList, me, GO_SUPPRESSION_DEVICE, 200.0f);
+            for (std::list<GameObject*>::const_iterator itr = _goList.begin(); itr != _goList.end(); itr++)
+                ((*itr)->AI()->DoAction(ACTION_DEACTIVATE));
```
Comment (Winfidonarleyan) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_broodlord_lashlayer.cpp:86]: brackets and can use range for

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**